integration/tests/neon_evm/test_revision.py

```diff
@@ -1003,3 +1027,239 @@ def send_transaction_steps(holder_account, accounts):
+            expected_tag=TAG_ACTIVE_STATE,
+        )
+
+        recipient_balance_before_trx2 = evm_loader.get_neon_balance(recipient.eth_address)
+        payer_bytes32 = neon_api_client.call_contract_get_function(
+            sender_with_tokens, revision_revert_contract, "getPayer()"
+        )
+        payer = bytes32_to_solana_pubkey(payer_bytes32)
+
+        instruction = make_CreateAssociatedTokenIdempotent(
+            payer, sender_with_tokens.solana_account_address, Pubkey.from_string(environment.spl_neon_mint)
+        )
+        serialized_instructions = serialize_instruction(ASSOCIATED_TOKEN_ACCOUNT_PROGRAM_ID, instruction)
+
+        signed_tx2 = make_contract_call_trx(
+            evm_loader,
+            sender_with_tokens,
+            revision_revert_contract,
+            "transferNeonAndCallSolana(uint64,bytes,uint256,address)",
+            [2039280, serialized_instructions, amount, recipient.eth_address],
+            value=amount,
+        )
+
+        emulate_result = neon_api_client.emulate_contract_call(
+            sender_with_tokens.eth_address.hex(),
+            revision_revert_contract.eth_address.hex(),
+            "transferNeonAndCallSolana(uint64,bytes,uint256,address)",
+            [2039280, serialized_instructions, amount, recipient.eth_address],
+            value=hex(amount),
+        )
+        accounts_from_emulation2 = [Pubkey.from_string(item["pubkey"]) for item in emulate_result["solana_accounts"]]
+
+        evm_loader.write_transaction_to_holder_account(signed_tx2, holder2, operator_keypair)
+
+        # iterative execution of trx2
+        resp2 = evm_loader.execute_transaction_steps_from_account(
+            operator_keypair, treasury_pool, holder2, accounts_from_emulation2
+        )
+        check_transaction_logs_have_text(solana_client=sol_client, trx=resp2, text="exit_status=0x11")
+
+        payer_info = evm_loader.get_account_info(payer, commitment=Confirmed)
+        assert payer_info.value is None
+
+        recipient_balance_after_trx2 = evm_loader.get_neon_balance(recipient.eth_address)
+        assert recipient_balance_after_trx2 == recipient_balance_before_trx2 + amount
+
+        resp1 = evm_loader.execute_transaction_steps_from_account(
+            operator_keypair, treasury_pool, holder1, accounts_from_emulation1, check_invalid_revision=True
+        )
+        check_transaction_logs_have_text(solana_client=sol_client, trx=resp1, text="exit_status=0x11")
+
+        recipient_balance_after_trx1_rerun = evm_loader.get_neon_balance(recipient.eth_address)
+        assert recipient_balance_after_trx1_rerun == recipient_balance_before_trx2 + amount + 10 * amount
+        assert evm_loader.get_neon_balance(revision_revert_contract.eth_address) == 0
+
+        balance_account_revision_after = evm_loader.get_balance_account_revision(
+            revision_revert_contract.balance_account_address
+        )
+        assert balance_account_revision_after == balance_account_revision
```

Comment:
strange
you shoud change balance account in second trx to see the revision is changed



now you your value in trx == how much tokens you send to recipient

Reply:
Idk, I did check this revision and I also expected to have it changed. May be we should ask Anton?